CRM/Utils/Recent.php

```diff
@@ -179,12 +179,16 @@ private static function getTitle($entityType, $entityId) {
     $labelField = CoreUtil::getInfoItem($entityType, 'label_field');
     $title = NULL;
     if ($labelField) {
-      $record = civicrm_api4($entityType, 'get', [
-        'where' => [['id', '=', $entityId]],
-        'select' => [$labelField],
-        'checkPermissions' => FALSE,
-      ], 0);
-      $title = $record[$labelField] ?? NULL;
+      try {
+        $record = civicrm_api4($entityType, 'get', [
+          'where' => [['id', '=', $entityId]],
+          'select' => [$labelField],
+        ])->first();
+      }
+      catch (\API_Exception $e) {
+        $record = [];
+      }
+      $title = $record[$labelField] ?? ts('Unknown record');
     }
     return $title ?? (CoreUtil::getInfoItem($entityType, 'label_field'));
```

Comment:
@colemanw Hrm, did I position the `ts('Unknown record')` poorly? I didn't actually read this line while drafting the other bits.

I guess L193 is saying: "If we didn't figure out a `$title`, then just use the name of the entity-type ('Contact', 'Activity', etc)." But as written, maybe it's kinda hard to get a `NULL` value of `$title`? (They're both fallbacks - just different.)

I haven't tested all these edges... but perhaps both L191 and L193 should be changing? eg

```php
      $title = $record[$labelField] ?? NULL;
    }
    return $title ?? (CoreUtil::getInfoItem($entityType, 'title'));
```

Or

```php
      $title = $record[$labelField] ?? NULL;
    }
    return $title ?? ts('%1 #%2', [
      1 => CoreUtil::getInfoItem($entityType, 'title'),
      2 => $entityId,
    ]);
```

Reply:
@totten AFAIK the line I commented on is not reachable except *maybe* in odd circumstances such as the record having been deleted.

The logic here all feels a little incongruent because ostensibly this is a simple function to fetch a title, but it winds up being a permission checker too.

How about checking permissions before we call this function, and actually following the `$checkPermissions` parameter in doing so? Like this: #23876